junit5-extension/src/main/java/io/kroxylicious/testing/kafka/junit5ext/KafkaClusterExtension.java

```diff
@@ -1182,6 +1191,23 @@ static KafkaClusterProvisioningStrategy findBestProvisioningStrategy(
                 });
     }
 
+    // if the user sets TEST_CLUSTER_PREFERRED_EXECUTION_MODE, we want to prefer this mode if the user declares a generic KafkaCluster
+    private static Function<KafkaClusterProvisioningStrategy, Integer> preferStrategiesMatchingEnvVar() {
+        return x -> {
+            String preferredModeFromEnv = System.getenv().get(TEST_CLUSTER_PREFERRED_EXECUTION_MODE);
```

Comment:
The env var TEST_CLUSTER_EXECUTION_MODE is used by KafkaClusterFactory code path (which is not used by the KafkaClusterProvisioningStrategy).   I wondering if we should use the same environment variable rather than introducing a new one.

Reply:
sounds good, I think sharing the env var will be clear enough despite the subtleties

Reply:
going to add some docs to the front README with details about how to influence the provisioning mechanism for users.